### PR TITLE
refactor: Avoid `new`/`delete` in a number of places.

### PR DESCRIPTION
**Pull request by paulgessinger**

### CODE REVIEW

File: Core/include/Acts/Geometry/SurfaceArrayCreator.hpp [hunk 1]
```diff
@@ -130,23 +130,7 @@ class SurfaceArrayCreator {
...
-  /// @pre the pointers to the sensitive surfaces in the surfaces vectors all
-  /// need to be valid, since no check is performed
-  /// @param [in] gctx The gometry context for this building call
-  /// @param protoLayerOpt The proto layer containing the layer size
-  /// @param bTypePhi the binning type in phi direction (equidistant/arbitrary)
-  /// @param bTypeZ the binning type in z direction (equidistant/arbitrary)
-  /// @param transform is the (optional) additional transform applied
-  ///
-  /// @return a unique pointer a new SurfaceArray
+  /// SurfaceArrayCreator interface methonew SurfaceArray
```
Comment (andiwand) [inline Core/include/Acts/Geometry/SurfaceArrayCreator.hpp:133]: intentional?

Comment (paulgessinger) [reply Core/include/Acts/Geometry/SurfaceArrayCreator.hpp:133]: I don't know how that happened

Comment (paulgessinger) [reply Core/include/Acts/Geometry/SurfaceArrayCreator.hpp]: Reverted now.

**Review by andiwand — APPROVED** [review]

**Comment by andiwand** [comment]
ah build is still failing

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [6e3a9a683e66cb28169f15f8d8f6c8390e01651d]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46407799) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8575745)!

**Pull request merged**